### PR TITLE
Set Stripe api key directly in tests instead of using `allow`

### PR DESCRIPTION
**Pull request by andrewpbrett**
#### What? Why?

Closes https://github.com/openfoodfoundation/openfoodnetwork/issues/7357
See also https://github.com/openfoodfoundation/openfoodnetwork/pull/7327

This allows us to update the Stripe gem, which also clears the way for Rails 6.0

#### What should we test?

Green build.

#### Release notes

Updated tests to be compatible with latest Stripe gem

Changelog Category: User facing changes | Technical changes

### CODE REVIEW

**Review by luisramos0 — APPROVED** [review]
ah, nice, this was already blocking the stripe upgrade :+1:

File: spec/controllers/admin/stripe_connect_settings_controller_spec.rb [hunk 1]
```diff
@@ -30,7 +30,7 @@
 
       context "when a Stripe API key is not set" do
         before do
-          allow(Stripe).to receive(:api_key) { nil }
+          Stripe.api_key = nil
```
Comment (Matt-Yorkley) [inline spec/controllers/admin/stripe_connect_settings_controller_spec.rb:33]: Ah, in this case I think this setting will be persisted between tests...

Comment (andrewpbrett) [reply spec/controllers/admin/stripe_connect_settings_controller_spec.rb:33]: Hm, even setting it back after that spec runs still results in failures elsewhere in the build. I'm going to revert that commit for now and hopefully get the build back to green.

**Comment by andrewpbrett** [comment]
This is now included the Rails 6.0 PR, closing.

**Pull request closed without merging**